src/js/common/schemaform/actions.js

```diff
@@ -59,46 +58,86 @@ export function submitForm(formConfig, form) {
       event: `${formConfig.trackingPrefix}-submission`,
     });
 
-    const fetchOptions = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-        'X-Key-Inflection': 'camel'
-      },
-      body
-    };
-
-    const userToken = sessionStorage.userToken;
-    if (userToken) {
-      fetchOptions.headers.Authorization = `Token token=${userToken}`;
-    }
+    const promise = new Promise((resolve, reject) => {
+      const req = new XMLHttpRequest();
 
-    return fetch(`${environment.API_URL}${formConfig.submitUrl}`, fetchOptions)
-    .then(res => {
-      if (res.ok) {
-        window.dataLayer.push({
-          event: `${formConfig.trackingPrefix}-submission-successful`,
+      req.open('POST', `${environment.API_URL}${formConfig.submitUrl}`);
+      req.addEventListener('load', () => {
+        if (req.status >= 200 && req.status < 300) {
+          window.dataLayer.push({
+            event: `${formConfig.trackingPrefix}-submission-successful`,
+          });
+          // got this from the fetch polyfill, keeping it to be safe
+          const responseBody = 'response' in req ? req.response : req.responseText;
+          const results = JSON.parse(responseBody);
+          resolve(results);
+        } else {
+          const error = new Error(`vets_server_error: ${req.statusText}`);
+          Raven.captureException(error, {
+            extra: {
+              clientError: false,
+              statusText: req.statusText
+            }
+          });
```

Comment:
I ended up moving this out more and calling it from the promise's `catch` function. It's possible the way I had it before would miss errors in the promise callback function.